Muhammad Ramadhan

ram-a-dhan opened the pull request:
[ENV]
HOST=localhost
PORT=3000
DB_NAME=e_commerce_cms_test
DB_USER=postgres
DB_PASS=postgres

skrg udah ngga async timeout error

ram-a-dhan committed:
first commit

ram-a-dhan committed:
still testing; still getting async timeout error;

ram-a-dhan committed:
no more async timeout error;

ram-a-dhan committed:
finished register testing;

ram-a-dhan committed:
testing bcrypted password;

ram-a-dhan committed:
done w/ user test; just generated & migrated Product;

ram-a-dhan committed:
finished making test file & routes for Products but havent made the controller;

ram-a-dhan commented:
[env update]
HOST=localhost
PORT=3000
DB_NAME=e_commerce_cms_test
DB_USER=postgres
DB_PASS=postgres
JWT_SECRET=hacktiv8

sori kak baru ngeh kalo harus tambahin command initdb di package.json

Sursev07 commented on server/tests/user-login.test.js:
```diff
@@ -0,0 +1,74 @@
+    })
+})
+
+// afterAll(done => {
```
kenapa dicomment?

ram-a-dhan replied:
oh itu awalnya reset isi tabel di akhir test tapi aku ubah jadi resetnya di awal test & lupa hapus itu

Sursev07 commented on server/config/config.js:
```diff
@@ -0,0 +1,25 @@
+  "test": {
+    "username": process.env.DB_USER,
+    "password": process.env.DB_PASS,
+    "database": process.env.DB_NAME,
```
ini beda kan namanya sama db test?

ram-a-dhan replied:
iya kak

ram-a-dhan committed:
finished Read tests; havent touched Vue CLI & V-Router;

ram-a-dhan committed:
finished server CRUD test;

ram-a-dhan committed:
done with basic requirements except for the register feature;

ram-a-dhan committed:
deploying server to heroku;

ram-a-dhan committed:
forgot to add procfile;

ram-a-dhan committed:
add hashing to user seed;

ram-a-dhan committed:
deploying to firebase;

ram-a-dhan committed:
finished deploying front & back; API docs WIP;

ram-a-dhan commented:
Deploy Link:
- Firebase: https://tunggalika-3782a.firebaseapp.com
- Heroku: https://mighty-stream-85910.herokuapp.com

test accounts:
- username: "admin"; password: "admin";
- username: "user"; password: "user";

API Docs Link:
- Postman: ~~WIP~~ https://documenter.getpostman.com/view/10571856/SzS8u5y5

~~Coming Soon:~~ Added:
- Registration Feature

env update:
HOST=localhost
PORT=3000
DB_NAME_TEST=e_commerce_cms_test
DB_NAME_DEV=e_commerce_cms_dev
DB_USER=postgres
DB_PASS=postgres
JWT_SECRET=hacktiv8

ram-a-dhan committed:
added auth for registration;

ram-a-dhan committed:
misc edit for registration test;

ram-a-dhan committed:
reactivated app.listen;

ram-a-dhan committed:
pre-deploy testing;

ram-a-dhan committed:
finished all coding; next is API docs;

ram-a-dhan committed:
finished API docs;

ram-a-dhan committed:
started reworking server to include shopping cart;
finished redefining models & tables;
added superadmin in add-users seeder;

ram-a-dhan committed:
finished remapping routes;

ram-a-dhan committed:
finished making shoppingcart controllers;
finished reworking other controllers;
misc updates on other files to support new system;

ram-a-dhan committed:
fixed faulty routing for shoppingcart;
fixed shoppingcart controller logic errors;

ram-a-dhan committed:
polished controllers;

ram-a-dhan committed:
fixed buggy checkout controller;

ram-a-dhan committed:
fixed checkout bug;
ready to redeploy;
client-side isnt updated yet;

ram-a-dhan committed:
updated routings for CMS client;

ram-a-dhan committed:
improved dummy data;
reconfigured heroku DB credentials;
ready to redeploy;

ram-a-dhan committed:
server deployed; client deployed; deploy success;

ram-a-dhan committed:
Mengganti config js
